test/spectests/spectests.cpp

```diff
@@ -103,7 +121,25 @@ class test_runner
 
                 const auto registered_name = cmd.at("as").get<std::string>();
 
-                registered_names[registered_name] = module_name;
+                if (module_name == UnnamedModule)
```

Comment:
@gumb0 can you extend the smoke test with this case?

Reply:
added